### DISCUSSION TITLE
rvizでの/kachaka/front_camera/image_rawの画像表示について

### DISCUSSION THREAD

**Discussion by kohei0809** [pf-robotics/kachaka-api#57] [Q&A] [answered]
### 質問
ROS2からkachakaを使用する際に、/kachaka/front_camera/inage_rawのtopicをrvizで画像表示したいのですが、「No image」と表示されてしまいます。kachakaからのカメラ画像をrvizで表示する方法を教えていただきたいです。

### 環境について
MacBook Pro(M1チップ)から、Virtual BoxでUbuntu 22.04 LTSを使用しています。

### 情報
・READMEに記載されているように、kachaka_description/config/kachka.rvizを使用
・rvizではFrontCameraのstatusがOKとなっていて、QoSのReliabilityがBest Effortになっていることを確認
・kachakaの前カメラの隣についている白色のLEDトーチはついている
・「ros2 topic echo /kachaka/front_canera/image_raw」や「ros2 topic hz /kachaka/front_canera/image_raw」でトピックが送られていることは確認済みだが、hzで確認したfpsは1.0前後
・rvizを起動すると、マップや現在の位置などは表示されるが、ターミナルの方で「[ERROR] [1694851192.491584123] [rviz2]: Vertex Program:rviz/glsl120/indexed_8bit_image.vert Fragment Program:rviz/glsl120/indexed_8bit_image.flag GLSL link result: active samplers with a different type refer to the same texture image unit」と赤文字表示されている
・rvizを起動すると、ros2 topic echoやros2 topic hzが止まってしまう
・Jupiter Notebookでの/kachaka_api/python/demos/get_front_camera_continuous.ipynbやスマホアプリからkachakaのカメラ映像は確認することができる

よろしくお願いいたします。

**Comment by youtalk** [comment]
ご不便をおかけしております。
Virtual BoxのUbuntuのネットワークはホストPCとブリッジ接続でしょうか？ネットワーク周りが一番怪しそうです。

> ・「ros2 topic echo /kachaka/front_canera/image_raw」や「ros2 topic hz /kachaka/front_canera/image_raw」でトピックが送られていることは確認済みだが、hzで確認したfpsは1.0前後

10hz以上は出るはずなので、この時点でおかしそうです。`/kachaka/front_canera/image_raw/compressed`トピックの場合はどうなるでしょうか？こちらの方が圧縮画像である分、通信が軽いです。

**Reply by kohei0809** [reply, marked as answer]
アドバイスをもとに、ネットワーク周りを調査しました。
ブリッジ接続への変更とプロセッサー数を増やしたところ、`/kachaka/front_camera/image_raw`のfps約2まで上がり、`/kachaka/front_camera/image_raw/conpressed`は9まで上がりました。(ブリッジ接続のみだと、少し上昇)
また、rvizでも画像表示することができました。
ありがとうございました。

**Reply by youtalk** [reply]
解決できて良かったです！詳細な情報共有のおかげです。ご報告ありがとうございました。
